File: backend/models/Message.js
import mongoose, { Schema } from "mongoose";

const MessageSchema = new Schema(
  {
    sender: { type: Schema.Types.ObjectId, ref: "User", required: true },
    receiver: { type: Schema.Types.ObjectId, ref: "User", required: true },
    content: { type: String },
    messageType: {
      type: String,
      enum: ["text", "image", "document", "system", "video_call"],
      default: "text",
    },
    fileUrl: { type: String },
    fileName: { type: String },
    isRead: { type: Boolean, default: false },
    swapRequest: { type: Schema.Types.ObjectId, ref: "SwapRequest" },
  },
  { timestamps: true },
);

export default mongoose.model("Message", MessageSchema);

File: backend-spring/src/main/java/com/skillswap/backend/model/Message.java
package com.skillswap.backend.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.LocalDateTime;

@Data
@Document(collection = "messages")
public class Message {
    @Id
    @JsonProperty("_id")
    private String id;
    private String senderId;
    private String receiverId;
    private String content; // Replaces text
    private String messageType = "text"; // text, image, document, video_call, system
    private String fileUrl;
    private String fileName;
    private String swapRequestId;
    private boolean isRead = false;

    private boolean isSystemMessage = false;
    private LocalDateTime createdAt = LocalDateTime.now();
    private LocalDateTime updatedAt = LocalDateTime.now();
}
